src/components/tests/SourcesTree.spec.js

```diff
@@ -85,4 +86,622 @@ describe("SourcesTree", () => {
+          label: "Remove directory root"
+        }
+      ];
+      await component.instance().onContextMenu(event, rootDirectory);
+      expect(showMenu).toHaveBeenCalledWith(event, menuOptions);
+
+      expect(event.preventDefault).toHaveBeenCalled();
+      expect(event.stopPropagation).toHaveBeenCalled();
+
+      showMenu.mock.calls[0][1][0].click();
+      expect(props.setProjectDirectoryRoot).not.toHaveBeenCalled();
+      expect(props.clearProjectDirectoryRoot).toHaveBeenCalled();
+      expect(copyToTheClipboard).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("renderItem", () => {
+    const { component, props } = render();
+
+    it("should show icon for webpack item", async () => {
+      const item = createMockItem("webpack://", "webpack://");
+      const node = component
+        .instance()
+        .renderItem(item, 1, false, null, false, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show icon for angular item", async () => {
+      const item = createMockItem("ng://", "ng://");
+      const node = component
+        .instance()
+        .renderItem(item, 1, false, null, false, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show icon for moz-extension item", async () => {
+      const item = createMockItem("moz-extension://", "moz-extension://");
+      const node = component
+        .instance()
+        .renderItem(item, 1, false, null, false, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show icon for folder with arrow", async () => {
+      const item = createMockItem("folder", "folder", []);
+      const node = component
+        .instance()
+        .renderItem(item, 1, false, null, false, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show icon for folder with expanded arrow", async () => {
+      const item = createMockItem("folder", "folder", []);
+      const node = component
+        .instance()
+        .renderItem(item, 1, false, null, true, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show focused item for folder with expanded arrow", async () => {
+      const item = createMockItem("folder", "folder", []);
+      const node = component
+        .instance()
+        .renderItem(item, 1, true, null, true, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show source item with source icon", async () => {
+      const node = component
+        .instance()
+        .renderItem(itemOne, 1, false, null, false, { setExpanded: jest.fn() });
+      expect(node).toMatchSnapshot();
+    });
+
+    it("should show source item with source icon with focus", async () => {
+      const node = component
+        .instance()
+        .renderItem(itemOne, 1, true, null, false, { setExpanded: jest.fn() });
```

Comment:
it is unclear to me what each of these args means, can you think of a way to make it clearer to a reader?